audio-recorder.js

```diff
@@ -86,64 +85,59 @@ module.exports = class AudioRecorder {
     // Detect audio recording capabilities.
     // http://caniuse.com/#feat=stream
     // https://developer.mozilla.org/en-US/docs/Web/API/Navigator.getUserMedia
-    navigator.getUserMedia = navigator.getUserMedia ||
-      navigator.webkitGetUserMedia ||
-      navigator.mozGetUserMedia ||
-      navigator.msGetUserMedia;
-    this.canGetUserMedia = Boolean(navigator.getUserMedia);
-    console.log('Native deprecated navigator.getUserMedia API capability: ' +
-      this.canGetUserMedia);
-
-    // https://developer.mozilla.org/en-US/docs/Web/API/MediaDevices/mediaDevices.getUserMedia
-    this.canMediaDevicesGetUserMedia = false;
-    if (navigator.mediaDevices) {
-      navigator.mediaDevices.getUserMedia = navigator.mediaDevices.getUserMedia ||
-        navigator.mediaDevices.webkitGetUserMedia ||
-        navigator.mediaDevices.mozGetUserMedia;
-      this.canMediaDevicesGetUserMedia = Boolean(navigator.mediaDevices.getUserMedia);
-    }
-    console.log('Native navigator.mediaDevices.getUserMedia API capability:',
-      this.canMediaDevicesGetUserMedia);
-
-    // Detect MediaStream Recording
-    // It allows recording audio using the MediaStream from the above
-    // getUserMedia directly with a native codec better than Wave.
-    // http://www.w3.org/TR/mediastream-recording/
-    this.canUseMediaRecorder = Boolean(window.MediaRecorder);
-    console.log('Native MediaRecorder recording capability: ' +
-      this.canUseMediaRecorder);
-
-    // Web Audio API
-    // High-level JavaScript API for processing and synthesizing audio
-    // http://caniuse.com/#feat=audio-api
-    window.AudioContext = window.AudioContext ||
-      window.webkitAudioContext || window.mozAudioContext;
-    const canCreateAudioContext = Boolean(window.AudioContext);
-    console.log('Native Web Audio API (AudioContext) processing capability: ' +
-      canCreateAudioContext);
-
-    // Detect Cordova Media Recording
-    // It allows recording audio using the native bridge inside WebView Apps.
-    // Note that it may also require native playback when codecs were used for
-    // recording that are not yet supported in the WebView.
-    // https://github.com/apache/cordova-plugin-media/blob/master/doc/index.md
-    this.canUseCordovaMedia = Boolean(window.Media);
-    console.log('Cordova Media recording capability: ' +
-      this.canUseCordovaMedia);
-
-    if (!(this.canGetUserMedia || this.canUseCordovaMedia)) {
-      throw new Error(
-        'Some form of audio recording capability is required');
-    }
+    return getUserMedia({audio: true})
+      .then(() => {
+        this.canGetUserMedia = true;
+        return Promise.resolve();
```

Comment:
```js
return Promise.resolve();
```

Is the same as:

```js
return Promise.resolve(undefined);
```

Is the same as:

```js
return undefined;
```

Is the same as:

```js
return;
```

Is the same as:

```js
```

So it can be removed entirely.

Also, wouldn't this be better?

```js
return getUserMedia({audio: true})
  .then(() => true)
  .catch(() => false)
  .then(canGetUserMedia => {...});
```

In the current setup `this.canGetUserMedia` is assigned in this method, but not in the constructor. Its existence is ambiguous.